pako/index.d.ts

```diff
@@ -1,61 +1,84 @@
-// Type definitions for pako 0.2.8
+// Type definitions for pako 1.0.4
 // Project: https://github.com/nodeca/pako
-// Definitions by: Denis Cappellin <http://github.com/cappellin>
+// Definitions by: Denis Cappellin <https://github.com/cappellin>, Caleb Eggensperger <https://github.com/calebegg>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 export = Pako;
 export as namespace pako;
 
 declare namespace Pako {
+	export interface DeflateOptions {
+		level?: number;
+		windowBits?: number;
+		memLevel?: number;
+		strategy?: number;
+		dictionary?: any;
+		raw?: boolean;
+		to?: 'string';
+	}
+
+  export interface InflateOptions {
+		windowBits?: number;
+		raw?: boolean;
+		to?: 'string';
+	}
+
+	export type Data = Uint8Array | Array<number> | string;
+
 	/**
 	 * Compress data with deflate algorithm and options.
 	 */
-	export function deflate( data: Uint8Array | Array<number> | string, options?: any ): string;
+	export function deflate( data: Data, options: DeflateOptions & {to: 'string'} ): string;
+	export function deflate( data: Data, options?: DeflateOptions ): Uint8Array;
```

Comment:
No, I'm just trying to enforce that if `{to: 'string'}` is present, the return type is 'string'. [Playground link](http://www.typescriptlang.org/play/index.html#src=declare%20namespace%20Foo%20%7B%0D%0A%20%20%20%20interface%20Data%20%7B%20%7D%0D%0A%20%20%20%20interface%20DeflateOptions%20%7B%20%7D%0D%0A%20%20%20%20export%20function%20deflate(data%3A%20Data%2C%20options%3A%20DeflateOptions%20%26%20%7B%20to%3A%20'string'%20%7D)%3A%20string%3B%0D%0A%20%20%20%20export%20function%20deflate(data%3A%20Data%2C%20options%3F%3A%20DeflateOptions)%3A%20Uint8Array%3B%0D%0A%7D%0D%0A%0D%0Alet%20a%3A%20Uint8Array%20%3D%20Foo.deflate('a')%3B%0D%0Alet%20a2%3A%20Uint8Array%20%3D%20Foo.deflate('a'%2C%20%7B%20stuff%3A%20true%20%7D)%3B%0D%0Alet%20s%3A%20string%20%3D%20Foo.deflate('a'%2C%20%7B%20to%3A%20'string'%20%7D)%3B)